File: aprilflow-python-sdk/aprilflow/models/document.py
# Copyright 2026 April Software
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements. See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership. The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License. You may obtain a copy of the License at
#
#  http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied. See the License for the
# specific language governing permissions and limitations
# under the License.

from __future__ import annotations
from datetime import datetime
from enum import Enum
from pydantic import Field
from .base import AprilFlowModel

class DocumentStatus(str, Enum):
    CREATED = "Created"
    PROCESSED = "Processed"
    DELETED = "Deleted"

class DocumentType(str, Enum):
    ORIGINAL = "Original"
    MARKDOWN = "Markdown"

class Chunk(AprilFlowModel):
    collection_id: str | None = Field(None, alias="collectionId")
    chunk_id: str | None = Field(None, alias="chunkId")
    text: str | None = None
    start_line: int | None = Field(None, alias="startLine")
    end_line: int | None = Field(None, alias="endLine")
    score: float | None = None

class Document(AprilFlowModel):
    collection_id: str | None = Field(None, alias="collectionId")
    id: str | None = None
    document_type: DocumentType | None = Field(None, alias="documentType")
    creation_date: datetime | None = Field(None, alias="creationDate")
    status: DocumentStatus | None = None
    status_date: datetime | None = Field(None, alias="statusDate")
    file_name: str | None = Field(None, alias="fileName")
    file_size: int | None = Field(None, alias="fileSize")

class DocumentId(AprilFlowModel):
    collection_id: str = Field(alias="collectionId")
    document_id: str = Field(alias="documentId")

class DocumentItem(AprilFlowModel):
    document: Document | None = None
    overview: str | None = None
    score: float | None = None
    chunks: list[Chunk] = []

class DocumentSearchRequest(AprilFlowModel):
    collection_id: str = Field(alias="collectionId")
    text: str
    max_result: int | None = Field(None, alias="maxResult")

File: aprilflow-java-sdk/src/main/java/aprilflow/sdk/document/Document.java
/*
 * Copyright 2026 April Software
 * 
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package aprilflow.sdk.document;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;

public final class Document
{
    private final String collectionId;
    private final String id;
    private final DocumentType documentType;
    private final Instant creationDate;
    private final DocumentStatus status;
    private final Instant statusDate;
    private final String fileName;
    private final int fileSize;

    @JsonCreator
    public Document(
        @JsonProperty("collectionId") String collectionId,
        @JsonProperty("id") String id,
        @JsonProperty("documentType") DocumentType documentType,
        @JsonProperty("creationDate") Instant creationDate,
        @JsonProperty("status") DocumentStatus status,
        @JsonProperty("statusDate") Instant statusDate,
        @JsonProperty("fileName") String fileName,
        @JsonProperty("fileSize") int fileSize
    )
    {
        this.collectionId = collectionId;

        this.id = id;

        this.documentType = documentType;

        this.creationDate = creationDate;

        this.status = status;

        this.statusDate = statusDate;

        this.fileName = fileName;

        this.fileSize = fileSize;
    }

    public String getCollectionId()
    {
        return collectionId;
    }

    public String getId()
    {
        return id;
    }

    public DocumentType getDocumentType()
    {
        return documentType;
    }

    public Instant getCreationDate()
    {
        return creationDate;
    }

    public DocumentStatus getStatus()
    {
        return status;
    }

    public Instant getStatusDate()
    {
        return statusDate;
    }

    public String getFileName()
    {
        return fileName;
    }

    public int getFileSize()
    {
        return fileSize;
    }
}
